### ISSUE TITLE
Bubble-themed chat

### ISSUE THREAD

**Issue by nukeador** [vector-im/riotX-android#979]
Once thing that I miss the most about RiotX is the ability to display the chats with a style/theme that is more aligned with other mobile messaging apps.

A bubble-themed style is easier to the eyes and cleaner

![imagen](https://user-images.githubusercontent.com/531906/74167880-37195a80-4c29-11ea-9fd7-b1b5e76dea8f.png)

Vs the current IRC/Slack style:

![imagen](https://user-images.githubusercontent.com/531906/74167941-4dbfb180-4c29-11ea-93d0-33fdb7c5c014.png)

Maybe this could be an option people can switch, but most people I've been talking about expected a more familiar bubble-themed style by default on mobile.

**Comment by andycuccaro** [comment]
I think even something "not-so-bubbly", like the ones used by the stock SMS app in AOSP. I feel like the more squared corners don't waste as much screen real estate as the more rounded corners do. 
Example: 
![Screenshot_20200210-174921_Messaging](https://user-images.githubusercontent.com/54337672/74188860-143d7500-4c2e-11ea-9a14-6cb457f27117.png)

**Comment by irvin** [comment]
or just implement theming feature and let the community do the rest.

**Comment by nukeador** [comment]
Also, for some reason, I like the ability to identify my own messages since they are highlighted in a different color and use less space because I don't need to see my avatar.

**Comment by fbruetting** [comment]
I really like the current style, you get a lot of information onto the screen, but still there's enough space for logical separation. Especially the second screenshot uses way too much whitespace, so that you just get very little information on already small screens.

Matrix often is used for "real"/professiinal information exchange (like Slack) - that's another use-case than WhatsApp-like smalltalk to me. Current style fits both, bubble-style more the last example. 🤔

**Comment by bpcurse** [comment]
> Also, for some reason, I like the ability to identify my own messages since they are highlighted in a different color and use less space because I don't need to see my avatar.

xref: Highlighting messages - Issue at Riot Web
https://github.com/vector-im/riot-web/issues/12306

**Comment by jwsp1** [comment]
definitely a must-have for a future update. But for the moment the devs should still concentrate on adding basic functions. Anyways, hope to see this feature arrive soon, too

**Comment by SpiritCroc** [comment]
I created a fork of Element which features message bubbles (based on the ones found in the AOSP messaging app): https://github.com/SpiritCroc/SchildiChat-android
Since I didn't fully understand the code-generation that's going on in the relevant parts (in particular, I didn't see how I could best re-use the same code for different layouts), I don't think that it's suitable for including in Element (yet?), but I figured in case anybody wants to look into it, it might be a starting point.

**Comment by Hobby-Student** [comment]
I know many people not wanting to use Element because it does not look like the things they know. They compare it to WhatsApp... (you can like this comparison or not)
I use matrix (Element as primary client) as a tool in my daily business and try to introduce it to my private life too. But the feedback I get is mostly "unusable" because of the layout. Some would love to get away from centralized WhatsApp, if the usability (layout) would be similar.
I don't know if the community theming (@irvin) would be much of work to integrate, but it sounds really good.

@SpiritCroc very nice what you did.

**Comment by SpiritCroc** [comment]
I have not yet seen any theming solution on Android yet, which would be powerful enough for message bubbles, i.e. which allows to change layouts. That's usually for a good reason too, since bad themes could easily crash the app or even mislead users to do other actions than they want, if themes have full control over layouts.
To allow for bubble themes, I see the need to already have some basic message bubble implementation available in the app, which could be controlled with user themes.

**Comment by ski7777** [comment]
+1

**Comment by TheTrainor** [comment]
When I try to convince people to use Matrix/Element, the fact the UI doesn't more closely align with every other app they are used to has been one of the most-cited reasons for not making the switch.

I can definitely understand people who prefer the current view in big multi-person chat groups. But for the average 1:1 conversation it seems to be sub-optimal.

My personal preference would be to default to the familiar bubble-like chat for 1:1 -> 1:5 (?) person chats, any more than that and switch to the IRC style display.

I know I can suggest alternative clients but it's a hard sell when my mostly computer-illiterate friends go to the app store page for e.g. FluffyChat and see "1000 downloads"... they'll think it's a scam.

Just my 2c.

**Comment by arhayati** [comment]
+1 on the issue. Optional chat bubbles can satisfy all. I try to convince people to come to Matrix instead of other messengers and one of the major disadvantages is the UI of element as the most complete client.

**Comment by faaaaabi** [comment]
@SpiritCroc, nice job. I highly appreciate the bubble style in SchildiChat. How about joining force with the Element people in providing switchable timeline styles for Element Android?

**Comment by SpiritCroc** [comment]
> @SpiritCroc, nice job. I highly appreciate the bubble style in SchildiChat. How about joining force with the Element people in providing switchable timeline styles for Element Android?

If the Element developers are interested, I'm open to rework my implementation in order to get it in a state suitable for a PR (as the layout evolved over time, there are many commits to consider, which currently would not merge nicely). Without any statement from their side (design team?), I can't justify this effort right now.

**Comment by SimonBrandner** [comment]
@SpiritCroc We are working on this in Element Web/Desktop. The team has confirmed that they'd be interested in having this feature. I am not sure if they meant it in a general sense or just for Element Web. 

@nadonomy is the team interested in this on Android? I know that the team has a lot of work so this may not be possible right now. But I think eventually this would be necessary for feature parity with Element Web/Desktop

**Comment by nadonomy** [comment]
> @nadonomy is the team interested in this on Android? I know that the team has a lot of work so this may not be possible right now. But I think eventually this would be necessary for feature parity with Element Web/Desktop

Yes for sure. Before turning on in production there's a lot of things we'd need to figure out, things like:

 - Whether they should be enabled for DMs by default
 - Settings to expose per account, or per room
 - How this interacts with spaces
 - General UI styling

We would need to do some testing & research to find the right answers to the above and more in order to enable by default on production, but definitely would appreciate & benefit from a message bubble feature with a simple account level labs flag or setting to set to test the layout in general.

**Comment by jwsp1** [comment]
> If the Element developers are interested, I'm open to rework my implementation in order to get it in a state suitable for a PR (as the layout evolved over time, there are many commits to consider, which currently would not merge nicely). Without any statement from their side (design team?), I can't justify this effort right now.

Just saw #2804, don't know when it is going to be implemented but it would be nice if message bubbles had hollow/filled tick in DM and (additional) user-icons on rooms, just like in the linked issue.

**Comment by jwsp1** [comment]
Looks like iOS got a PR for chat bubbles https://github.com/vector-im/element-ios/pull/4150.
With SchildiChat, Element Android theoretically supports bubbles as well. Why can't the work of SchildiChat be implemented in Element Android as an experimental option?

**Comment by SpiritCroc** [comment]
> Looks like iOS got a PR for chat bubbles [vector-im/element-ios#4150](https://github.com/vector-im/element-ios/pull/4150).
> With SchildiChat, Element Android theoretically supports bubbles as well. Why can't the work of SchildiChat be implemented in Element Android as an experimental option?

Currently, I'm lacking the time to get it ready for creating a pull request. If somebody else wants to help by collecting all relevant parts from SchildiChat as a starting point to apply on top of the current Element codebase, that would also be fine with me :)

**Comment by piersonleo** [comment]
Hi, is there any update on this? We'd love to have bubble-themed chat option as it's more familiar for our users.

**Comment by kojid0** [comment]
> Hi, is there any update on this? We'd love to have bubble-themed chat option as it's more familiar for our users.

It was added to the "Now" section in the public roadmap, so development seems to start (soon): https://github.com/vector-im/roadmap/projects/1#column-11563017

**Comment by daniellekirkwood** [comment]
Good news! We're currently looking at this feature. Thanks all for your feedback. 

Please keep it coming so that we can continue to iterate on our current planned solution.

**Comment by Flam3z** [comment]
Another idea could be the "discord route" where the final chat theme will stay but have the IRC option as a toggle within settings then of course on top of that, theme tweaks that (hopefully) do not break the UI and crash the app (i.e. changing colors)

**Comment by brevilo** [comment]
A note on bubbles as global setting vs. room setting: https://github.com/vector-im/roadmap/issues/30#issuecomment-1033521558
